### PR TITLE
[DPE-4941]: Get rid of legacy relations

### PR DESCRIPTION
**Pull request by Gu1nness**
## Issue

 * We should remove the old legacy relations code

## Solution
 * Remove the old legacy code.

### CODE REVIEW

**Comment by Gu1nness** [comment]
Should be merged after #440 so internal versions are good

**Review by delgod — DISMISSED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -398,7 +391,7 @@ def _on_install(self, event: InstallEvent) -> None:
 
         # Construct the mongod startup commandline args for systemd and reload the daemon.
         update_mongod_service(
-            auth=auth,
+            auth=True,
```
Comment (MiaAltieri) [inline src/charm.py]: IIRC the only scenario in which we started `mongod` service without auth, was when we were using legacy relations.

Would you mind updating this function to not using this parameter anymore?

File: tests/integration/backup_tests/test_backups.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ async def test_blocked_incorrect_conf(ops_test: OpsTest, github_secrets) -> None
     await helpers.set_credentials(ops_test, github_secrets, cloud="AWS")
 
     # wait for both applications to be idle with the correct statuses
-    ops_test.model.wait_for_idle(apps=[S3_APP_NAME], status="active")
+    await ops_test.model.wait_for_idle(apps=[S3_APP_NAME], status="active")
```
Comment (MiaAltieri) [inline tests/integration/backup_tests/test_backups.py]: ugh ty

**Review by MiaAltieri — CHANGES_REQUESTED** [review]
Great work here Neha! One suggestion in one of my comments and a few more here.

Looks like there are still a few remaining references to legacy/obsolete, mind taking care of those?
```
grep -rn legacy lib/charms/mongodb/v1/mongodb_provider.py
grep -rn legacy CONTRIBUTING.md
grep -rni obsolete src/config.py 
```

File: src/charm.py (hunk 1 above)
Comment (Gu1nness) [reply src/charm.py]: Done!

**Review by Mehdi-Bendriss — DISMISSED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]

**Pull request merged**